src/bundle.c

```diff
@@ -424,12 +422,16 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	/* step 1: check bundle args are valid if so populate subs struct */
 	ret = add_subscriptions(bundles, subs, current_version, mom, 0);
 
-	if (ret) {
-		if (ret == 1) {
-			printf("bundle(s) already installed, exiting now\n");
-		}
+	switch (ret)
+	{
+	case 0: /* no errors, but nothing added */
+		printf("bundle(s) already installed, exiting now\n");
+		/* fall through */
+	case add_sub_ERR: /* error of some kind */
+	case add_sub_ERR|add_sub_NEW: /* error of some kind but we did add something */
 		ret = EBUNDLE_INSTALL;
 		goto out;
+	default: break;
```

Comment:
I hope not! The "default: break;" is in there to be sure that the switch has a case for every value that can be returned from add_subscriptions. Currently there are 4 possible bit patterns 00, 01, 10, 11. Of these 01 and 11 indicate that some error in the processing happened, e.g. failed to download a manifest. these will cause it to not install bundles.

00 is what you get if nothing was added, again this cause it to not install anything, but that is OK as there is nothing to install.

10 is the case handled by the "default:", it will break out of the switch statement and  continue to the set_subscription_versions call after it.

However you have drawn my attention to me not updating the '00' case message which was the point of the patch! Thanks.